update openshift install docs and example (#884)
DataDog/datadog-operator: docs/install-openshift.md

![Datadog Operator in the OperatorHub](assets/operatorhub.png)

Installation includes the creation of a `ServiceAccount` called `datadog-agent-scc` that is bound to two default OpenShift `SecurityContextConstraints`.

3. Specify the namespace to install the Datadog Operator in, you can use `datadog` if you previously created the project or an existing one, such as `openshift-operators`:

![Deploy the operator in the datadog namespace](assets/datadognamespace.png)

> **Note**: Prior to version 1.0, multiple `InstallModes` were supported in the `ClusterServiceVersion` (see the [OLM operator install doc][3] as a reference). Due to the introduction of the conversion webhook in 1.0, only the `AllNamespaces` `InstallMode` [is supported][4] going forward.

Reviewer: ```suggestion
**Note**: Prior to version 1.0, multiple `InstallModes` were supported in the `ClusterServiceVersion` (see the [OLM operator install doc][3] as a reference). Due to the introduction of the conversion webhook in 1.0, only the `AllNamespaces` `InstallMode` [is supported][4] in versions later than 1.0.
```

## Deploy the Datadog Agent with the Operator

> After deploying the Datadog Operator, create a `DatadogAgent` resource that triggers a deployment of the Datadog Agent in your OpenShift cluster. The Agent is deployed as a `Daemonset`. It is recommended to use the Cluster Agent to manage cluster-level monitoring, and this will also be deployed by default.

Reviewer: ```suggestion
After deploying the Datadog Operator, create a `DatadogAgent` resource that triggers a deployment of the Datadog Agent in your OpenShift cluster. The Agent is deployed as a `Daemonset`. Datadog recommends that you use the Cluster Agent to manage cluster-level monitoring, which will automatically be deployed by default.
```

> **Note**: In Datadog Operator versions `1.1.0` and above, the Conversion Webhook is **disabled** by default, and can be enabled with the command argument `--webhookEnabled`. In version `1.0.3` of the Datadog Operator, listing the Conversion Webhook is **enabled** by default. The conversion allows a smooth transition from the `v1alpha1` (deprecated) `DatadogAgent` CRD to `v2alpha1`.

Reviewer: ```suggestion
**Notes**: 
- In Datadog Operator versions `1.1.0` and later, the conversion webhook is **disabled** by default. To enable the webhook, use the command argument `--webhookEnabled`. 
- In Datadog Operator version `1.0.3`, the conversion webhook is **enabled** by default. The conversion allows a smooth transition from the `v1alpha1` (deprecated) `DatadogAgent` CRD to `v2alpha1`.
```

1. Create a Kubernetes secret with your API and App keys:

   ```shell
   oc create secret generic datadog-secret -n datadog --from-literal api-key=<DATADOG_API_KEY> --from-literal app-key=<DATADOG_APP_KEY>
   ```

>    Replace `<DATADOG_API_KEY>` and `<DATADOG_APP_KEY>` with your [Datadog API][5] and [Application keys][6]

Reviewer: ```suggestion
   Replace `<DATADOG_API_KEY>` and `<DATADOG_APP_KEY>` with your [Datadog API][5] and [Application keys][6].
```

![Datadog Operator in the OperatorHub](assets/operatorhub.png)

Installation includes the creation of a `ServiceAccount` called `datadog-agent-scc` that is bound to two default OpenShift `SecurityContextConstraints`.

3. Specify the namespace to install the Datadog Operator in, you can use `datadog` if you previously created the project or an existing one, such as `openshift-operators`:

![Deploy the operator in the datadog namespace](assets/datadognamespace.png)

**Note**: Prior to version 1.0, multiple `InstallModes` were supported in the `ClusterServiceVersion` (see the [OLM operator install doc][3] as a reference). Due to the introduction of the conversion webhook in 1.0, only the `AllNamespaces` `InstallMode` [is supported][4] in versions 1.0 and later.

## Deploy the Datadog Agent with the Operator

After deploying the Datadog Operator, create a `DatadogAgent` resource that triggers a deployment of the Datadog Agent in your OpenShift cluster. The Agent is deployed as a `Daemonset`. Datadog recommends that you use the Cluster Agent to manage cluster-level monitoring, which will automatically be deployed by default.

**Notes**:

> - In Datadog Operator versions `1.1.0` and later, the conversion webhook is **disabled** by default. To enable the webhook, use the command argument `--webhookEnabled`.

Author: ```suggestion
- In Datadog Operator versions `1.1` and later, the conversion webhook is **disabled** by default. To enable the webhook, use the command argument `--webhookEnabled`.
```

            type: spc_t
            user: system_u
        serviceAccountName: datadog-agent-scc
      clusterAgent:
        serviceAccountName: datadog-agent-scc
        replicas: 2
  ```

Setting the `serviceAccountName` in the `nodeAgent` and `clusterAgent` `override` section ensures that these pods are associated with the necessary `SecurityContextConstraints` and RBACs.

3. Apply the Datadog Agent manifest:
   ```shell
   oc apply -f path/to/your/datadog-agent.yaml
   ```

> The Datadog Agent and Cluster Agent should now be running and collecting data to be viewed and alerted on in the Datadog web app.

Reviewer: ```suggestion
The Datadog Agent and Cluster Agent should now be running and collecting data. This data can be viewed and alerted on in Datadog.
```
